plugins/ui-update/commands.ts

```diff
@@ -1,22 +1,67 @@
 import { graphqlObservable } from "@dcos/data-service";
 import gql from "graphql-tag";
+import { delay, take } from "rxjs/operators";
+
 import { default as uiServiceSchema } from "#SRC/js/data/ui-update";
-import { delay } from "rxjs/operators";
+
+import { getAction$ } from "./streams";
+import { UIActions, UIActionType } from "./types/UIAction";
+
+const uiServiceActions$ = getAction$();
 
 function rollbackUI(delayMs: number = 45000) {
-  return graphqlObservable<{ resetDCOSUI: string | null }>(
+  uiServiceActions$.next({
+    type: UIActionType.Reset,
+    action: UIActions.Started,
+    value: {
+      message: ""
+    }
+  });
+  graphqlObservable<{ resetDCOSUI: string | null }>(
     gql`
       mutation {
         resetDCOSUI
       }
     `,
     uiServiceSchema,
     {}
-  ).pipe(delay(delayMs));
+  )
+    .pipe(
+      take(1),
+      delay(delayMs)
```

Comment:
I know, wrong PR, but why do we delay this?

Reply:
It's a workaround for the multi-master race condition. We will remove it once the race condition is resolved.